### PR TITLE
fix: detection of introspection query

### PR DESCRIPTION
**Pull request by rheaditi**
It incorrectly detects `__typename` due to `__type` being a substring. my bad </3

--

Unsure about the other ways of detecting introspection queries. Will look into it further.

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+/**
+ * Identify if the request is an introspection query type
+ * @param request {GraphQLRequestContext['request]}
+ */
+export const isIntrospectionRequest = (request: GraphQLRequestContext<any>['request']) => {
+  if (typeof request.query === 'string') {
+    return REGEX_INTROSPECTION_QUERY.test(request.query);
+  }
+  return false;
```
Comment (maddhruv) [inline src/utils.ts]:
```suggestion
  return typeof request.query === 'string' && REGEX_INTROSPECTION_QUERY.test(request.query)
```

A small refactor :P

**Comment by pritam248** [comment]
https://graphql.org/learn/introspection/

File: src/utils.ts (hunk 1 above)
Comment (rheaditi) [reply src/utils.ts]: No :P

Comment (rheaditi) [reply src/utils.ts]: </defiance>

**Pull request merged**